specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,509 @@
+above, it is also very difficult to diagnose issues when developers hit them and a large portion of
+developer cycles are spent diagnosing and troubleshooting problems in various API stacks instead of
+a single stack.
+
+**Registration**: The Windows SDK today provides AppId overloads to developers who can specify their
+own Identifier for Unpackaged apps. This overloaded AppId causes all sorts of problems including
+string size limits being hit, Personally Identifiable Information stored as part of AppIds, invalid
+Desktop Registrations etc. The WinApp SDK gets rid of the AppId overloads in the Public APIs and
+ensures that we don't hit any of the above issues.
+
+**Activation**: Setting up activation handlers using Background Triggers is challenging for
+developers because we don't have downlevel support for versions prior to 1903. Moreover, the steps
+for setting up activation handlers vary greatly for different app types: MSIX Vs Unpackaged Vs UWP.
+We will abstract away the activation technology and simplify this process for OS SKUs from RS5
+onwards.
+
+# Description
+
+The WinAppSDK AppNotification APIs enable all Desktop applications to display the same rich UI
+pop-ups that they are used to doing today minus all the complexities and overhead described above.
+Desktop applications include Unpackaged apps and Packaged win32 (MSIX Desktop Bridge, Sparse Signed
+Packages). The SDK implementation will ensure that all Notification scenarios will adhere to OS
+resource management policies like Power Saver, Network Attribution (reporting the amount of data an
+app uses), Group Policies, etc. The SDK will also abstract away the complexities of dealing with
+Notification delivery and related activations as much as possible freeing the developer to focus on
+other app related challenges.
+
+The App Notification API allows Packaged and Unpackaged apps to:
+
+-   Use common APIs to Register and Unregister Notifications.
+-   Use common APIs to display Cloud Sourced Notifications.
+-   Use common APIs to display Local Notifications.
+-   Use common APIs to Update/Delete Notifications.
+-   Use AppLifecycle constructs and event handlers to process invoke operations on behalf of the
+    user.
+
+# Examples
+
+## Registering and handling a notification
+
+Packaged and Unpackaged Desktop apps will need to register themselves to handle App Notifications.
+For Packaged apps, WinAppSDK will use the CLSID defined in the appx manifest to register the current
+process as the COM Server for Notification invokes. The process.exe declared in the manifest will
+have to be the same process calling into Register() for the API call to succeed. For Unpackaged
+apps, WinAppSDK will call into Shell libraries to determine appropriate DisplayName and and Icon.
+The current process.exe will be the COM Server for Notification invokes.
+
+```cpp
+
+int main()
+{
+    // Register an event handler before calling Register()
```

Comment:
describe this in the positive sense, describing what is desired instead of what is being avoided.
// To ensure all notification handling happens in this process instance register for
// NotificationInvoked before calling Register(). Without this a new process will
// be launched to handle the notification.